processor/src/main/java/org/mapstruct/ap/spi/CaseEnumTransformationStrategy.java

```diff
@@ -0,0 +1,53 @@
+package org.mapstruct.ap.spi;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * Applies case transformation to the source enum
+ *
+ * @author jpbassinello
+ *
+ * @since 1.5
+ */
+public class CaseEnumTransformationStrategy implements EnumTransformationStrategy {
+
+    private static final String UPPER = "upper";
+    private static final String LOWER = "lower";
+    private static final String CAPITAL = "capital";
+
+    @Override
+    public String getStrategyName() {
+        return "case";
+    }
+
+    @Override
+    public String transform(String value, String configuration) {
+        switch (configuration.toLowerCase()) {
+            case UPPER:
+                return value.toUpperCase();
+            case LOWER:
+                return value.toLowerCase();
```

Comment:
Using `Locale.ROOT` when doing `toLowerCase` and `toUpperCase`

Reply:
Fixed.